src/main/java/seedu/address/storage/JsonAdaptedUuid.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.storage;
+
+import java.util.UUID;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Jackson-friendly version of {@link java.util.UUID}.
+ */
+class JsonAdaptedUuid {
+
+    public static final String MESSAGE_NULL_UUID = "UUID is null";
+
+    private final String uuid;
+
+    /**
+     * Constructs a {@code JsonAdaptedUuid} with the given {@code uuid}.
+     */
+    @JsonCreator
+    public JsonAdaptedUuid(String uuid) {
+        this.uuid = uuid;
+    }
+
+    /**
+     * Converts a given {@code Uuid} into this class for Jackson use.
+     */
+    public JsonAdaptedUuid(UUID source) {
+        uuid = source.toString();
+    }
+
+    @JsonValue
+    public String getUuid() {
+        return uuid;
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted uuid object into the model's {@code Uuid} object.
+     *
+     * @throws IllegalValueException if adapted uuid is null.
+     */
+    public UUID toModelType() throws IllegalValueException {
+        if (uuid == null) {
+            throw new IllegalValueException(MESSAGE_NULL_UUID);
+        }
+        return UUID.fromString(uuid);
```

Comment:
Nice handling of null UUIDs. However, would it be better if a try-catch block is used on `UUID.fromString(uuid)` instead? Then, any kinds of wrong UUIDs can be handled.

Reply:
Was thinking that a guard clause would be sufficient (Since its likely the case that a uuid is either well defined (by using Java's UUID.randomUUID(), or null.

Any thoughts on whether a try catch is needed here?

Reply:
To add on, according to Java's UUID API, `UUID.fromString()` throws `IllegalArgumentException`. Perhaps you can use this alongside with your null guard clause.

Reply:
Resolved by adding try catch block